aleph/views/context.py

```diff
@@ -66,11 +116,23 @@ def setup_request():
     request._http_etag = None
     request._log_tags = {}
 
+    enable_authz(request)
+    enable_rate_limit(request)
+
 
 @blueprint.after_app_request
 def finalize_response(resp):
     """Post-request processing to set cache parameters."""
-    generate_request_log(resp)
+    # Compute overall request duration:
+    took = time.time() - request._begin_time
+
+    # Finalize reporting of the rate limiter:
+    if hasattr(request, 'rate_limit') and request.rate_limit is not None:
+        usage = request.rate_limit.update(amount=math.ceil(took))
```

Comment:
Is `amount=took` there to charge users more for heavier requests?

Reply:
Yes -- the specific issue is the `/tags` API still, which sometimes takes more than a second to execute.